tests/backends/base/test_creation.py

```diff
@@ -101,3 +101,24 @@ def test_circular_reference(self):
         obj_ref = ObjectReference.objects.get()
         self.assertEqual(obj.obj_ref, obj_ref)
         self.assertEqual(obj_ref.obj, obj)
+
+    def test_self_reference(self):
+        # serialize_db_to_string() and deserialize_db_from_string() handles
+        # self references.
+        obj_1 = ObjectSelfReference.objects.create(key='X')
+        obj_2 = ObjectSelfReference.objects.create(key='Y', obj=obj_1)
+        obj_1.obj = obj_2
+        obj_1.save()
+        # Serialize objects.
+        with mock.patch('django.db.migrations.loader.MigrationLoader') as loader:
+            # Mark backends app as migrated.
```

Comment:
Why did you remove my comment here? Without it, it is not immediately clear why this mark as migrated needs to happen at all?

>      # serialize_db_to_string only serializes apps that have migrations, so pretend that (only) we do

Reply:
Agreed, it's a valuable comment. Restored.